func_darwin.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+package purego
+
+import (
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/ebitengine/purego/internal/strings"
+)
+
+// RegisterLibFunc takes a pointer to a Go function representing the calling convention of the C function named name
+// found in the shared object provided by handle.
+// fptr will be set to a function that when called will call the C function given by name with the
+// parameters passed in the correct registers and stack.
+//
+// A panic is produced if the name symbol cannot be found in handle or if the type is not a function
+// pointer or if the function returns more than 1 value.
+//
+// These conversions describe how a Go type in the fptr will be used to call
+// the C function. It is important to note that there is no way to verify that fptr
+// matches the C function. This also holds true for struct types where the padding
+// needs to be ensured to match that of C; RegisterLibFunc does not verify this.
+//
+// Type Conversions (Go => C)
+//
+//	string => char*
+//	bool => _Bool
+//	uintptr => uintptr_t
+//	uint => System Dependent
```

Comment:
`uint32_t` or `uint64_t`? Or just `uint`?

Reply:
As `unsigned int` is 32bit in most environments, doesn't this lose information...?

Reply:
True. I'll go with what you first said

Reply:
Thanks!